### PR TITLE
Amr Logical Agent Implementation

### LINKED ISSUE #11: Refactor Logical Agent Interface for AMR
The old implementation can be found in the AgvLogicalNs3 class. 

Now we want to implement the AmrLogicalAgent, derived from the LogicalAgent interface.

### PR DESCRIPTION
**Pull request by janagoe**
Closes #11

# Proposed Changes
Implementing the AmrLogicalAgent based on the LogicalAgent interface

# Definition of Done (optional)
- [x] Feature is implemented
- [x] No known bugs that stops the correct execution
- [x] CI / CD pipeline passed
- [x] The code guidelines were followed
- [ ] Unit and / or Integration tests for the new feature
- [ ] New feature was added to the documentation

### CODE REVIEW

File: daisi/src/cpps/logical/amr/amr_logical_agent.cpp [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  ns3::Address physical_address;
+  socket_of_physical_->GetSockName(physical_address);
+
+  ns3::InetSocketAddress i_physical_address = ns3::InetSocketAddress::ConvertFrom(physical_address);
+  std::string physical_asset_ip = daisi::getIpv4AddressString(i_physical_address.GetIpv4());
+  uint16_t physical_asset_port = i_physical_address.GetPort();
+
+  // retrieve logical info
+  ns3::Address logical_address;
+  socket_of_physical_->GetSockName(logical_address);
```
Comment (ltoenning) [inline daisi/src/cpps/logical/amr/amr_logical_agent.cpp:214]: Maybe a typo. To get the address of the logical, this should be the listening socket.

File: third_party/Catch2
Comment (ltoenning) [inline third_party/Catch2:1]: This is a downgrade from v3.3.2 to v3.2.1. Is this intentional? :smile:

File: solanet/third_party/cereal
Comment (ltoenning) [inline solanet/third_party/cereal:1]: This is a version upgrade. Is this intentional?

**Review by ltoenning — CHANGES_REQUESTED** [review]

File: daisi/src/cpps/logical/amr/amr_logical_agent.cpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  } else {
+    throw std::runtime_error("AmrDescription attempted to be set twice.");
+  }
+}
+
+void AmrLogicalAgent::processMessageAmrStatusUpdate(const AmrStatusUpdate &status_update) {
+  current_position_ = status_update.getPosition();
+  current_state_ = status_update.getState();
+
+  checkSendingNextTask();
```
Comment (tseeland) [inline daisi/src/cpps/logical/amr/amr_logical_agent.cpp:113]: Will you send the next task if the previous task is not finished yet? `AmrStatusUpdate` does not send information about the task.

Comment (tseeland) [inline daisi/src/cpps/logical/amr/amr_logical_agent.cpp:113]: Will you send the next task if the previous task is not finished yet? `AmrStatusUpdate` does not send information about the task.

**Review by tseeland — COMMENTED** [review]
All in all just minor questions. Folder structure is another issue for another time.

**Review by tseeland — APPROVED** [review]

**Review by tseeland — APPROVED** [review]
Nothing changed since last approval.

**Review by ltoenning — APPROVED** [review]

**Pull request merged**